### PR TITLE
Added functional tests for Rules Checker

### PR DESCRIPTION
**Pull request by ValentinTyhonov**

### CODE REVIEW

File: src/functionaltests/resources/rules/inputData.json [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            "StartEvent": "NO",
+            "IdentifyRules": "links | [?type=='SUBJECT'].target",
+            "MatchIdRules": {
+                "_id": "%IdentifyRules_objid%"
+            },
+            "ExtractionRules": "{confidenceLevels :[{ eventId:meta.id, time:meta.time, name:data.name, value:data.value}]}",
+            "ArrayMergeOptions": ""
+        }
+    ],
+    "listEventsJson": [
```
[the remark this answers is not in the document]
Comment (ValentinTyhonov) [reply src/functionaltests/resources/rules/inputData.json]: Yeah, you are right.

File: src/functionaltests/java/com/ericsson/ei/rules/RuleCheckSteps.java [hunk 2]
```diff
@@ -56,9 +58,12 @@ public void make_a_POST_request_with_JMESPath_rule_and_JSON_object_to_the_REST_A
...
 
-    @When("^make a POST request with list of JMESPath rules and list of JSON objects \"([^\"]*)\" to the REST API \"([^\"]*)\"$")
-    public void make_a_POST_request_with_list_of_JMESPath_rules_and_list_of_JSON_objects_to_the_REST_API(String requestBodyFileName, String endpoint) throws Throwable {
-        String requestBody = FileUtils.readFileToString(new File(TEST_RESOURCES_PATH + requestBodyFileName), "UTF-8");
+    @When("^make a POST request with list of JMESPath rules \"([^\"]*)\" and list of JSON objects \"([^\"]*)\" to the REST API \"([^\"]*)\"$")
+    public void make_a_POST_request_with_list_of_JMESPath_rules_and_list_of_JSON_objects_to_the_REST_API(String rulesFileName, String eventsFileName, String endpoint) throws Throwable {
+        String rules = FileUtils.readFileToString(new File(TEST_RESOURCES_PATH + rulesFileName), "UTF-8");
+        String events = FileUtils.readFileToString(new File(TEST_RESOURCES_PATH + eventsFileName), "UTF-8");
+        String requestBody = new JSONObject().put("listRulesJson", new JSONArray(rules)).put("listEventsJson", new JSONArray(events)).toString();
+        System.out.println(requestBody);
```
Comment (tobiasake) [inline src/functionaltests/java/com/ericsson/ei/rules/RuleCheckSteps.java]: Remove system out println or use Logger.

File: src/functionaltests/resources/rules/environmentDisabledResponse.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "message": "Test environment is not enabled. Please use the test environment for this execution"
```
Comment (tobiasake) [inline src/functionaltests/resources/rules/environmentDisabledResponse.json]: These Error messages will probably be changed in a coming story. Can't we just check the HTTP Error code for the moment?

Comment (ValentinTyhonov) [reply src/functionaltests/resources/rules/environmentDisabledResponse.json]: ok, I will not check any error messages

File: src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java [hunk 4]
```diff
@@ -70,16 +73,20 @@ public void setRest(RestOperations rest) {
...
+            final String erUrl = URI.create(url.trim() + "/" + "{id}").normalize().toString();
+            LOGGER.debug("The URL to ER is: " + erUrl);
+
+            final Map<String, String> params = Collections.singletonMap("id", eventId);
+            LOGGER.trace("The ID parameter is set");
+            try {
+                final ResponseEntity<String> response = rest.getForEntity(erUrl, String.class, params);
+                LOGGER.trace("The response is : " + response.toString());
+            } catch (RestClientException e) {
+                LOGGER.error("Error occurred while executing REST GET to: " + erUrl + " for " + eventId, e);
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java]: These Errors (or Error informations) should be included in the ResponseEntity, if its returned in EI RestAPI responses.

Comment (tobiasake) [reply src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java]: This comment applies to the other errors in this file which returns Responses to RestAPI.

Comment (ValentinTyhonov) [reply src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java]: I have only added LOGGER.info("The URL to ER is not provided"); here.

Comment (tobiasake) [reply src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java]: The Comment was about including the Error in the ResponseEntiy,, not Logger,,, Logger should be kept,, but include error into the ResponseEntity if its good information to user for clearing out what failed in the RestAPI call, if this was executed by a RestAPI request.

Comment (tobiasake) [reply src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java]: Let it be and it will be handled by coming new story.

**Pull request merged**